src/parser/mod.rs

```diff
@@ -1224,6 +1224,17 @@ impl<'a> Parser<'a> {
                             body: Box::new(self.parse_expr()?),
                         }));
                     }
+                    Token::DoubleColon if self.dialect.supports_methods() => {
+                        let namespace = w.value;
+                        self.expect_token(&Token::DoubleColon)?;
+                        let name_with_namespace = match self.next_token().token {
+                            Token::Word(func_name) => {
+                                ObjectName(vec![Ident::new(format!("{}::{}", namespace, func_name.value))])
+                            },
+                            _ => return self.expected("identifier", self.peek_token())
+                        };
+                        Ok(self.parse_function(name_with_namespace)?)
```

Comment:
Can we move the body to a `self.parse_namespaced_method(w)?` or similar function? Also if we can add an example syntax to the function doc with the link to clarify the syntax